src/ir_passes/await.ml

```diff
@@ -671,14 +671,23 @@ and t_ignore_throw context exp =
          (LabelEnv.add Throw (Cont throw) context) in
      let e = fresh_var "e" T.catch in
      { (blockE [
-          funcD throw e (tupE[]);
+          funcD throw e t_exp;
         ]
         (c_exp context' exp (meta (T.unit) (fun v1 -> tupE []))))
        (* timer logic requires us to preserve any source location,
           or timer won't be initialized in compile.ml *)
        with at = exp.at
      }
 
+and t_ignore_throw context exp = t_on_throw context exp (tupE[])
+
+(* if self-call queue full: expire global timer soon and retry *)
+and t_timer_throw context exp =
+  t_on_throw context exp
+    (blockE [expD (primE
```

Comment:
Are we sure there aren't other reasons for the error than queue full?  If so, it might make make sense to test the error code of the error and only set the global timer when appropriate...

Reply:
Checking the Wasm, the only possibility is send failure.